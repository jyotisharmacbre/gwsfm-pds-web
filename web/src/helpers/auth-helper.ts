// @ts-ignore
import authentication from '@kdpw/msal-b2c-react';
import jwtDecode from 'jwt-decode';
import { DecodedToken } from '../models/decodedToken';

export const getDisplayName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.name;
};

export const getFirstName = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.given_name;
};

export const getDisplayEmail = () => {
  const decoded: DecodedToken = decodeToken();
  return decoded.email ?
    decoded.email :
    decoded.emails && decoded.emails.length > 0 ?
      decoded.emails[0] :
      '';
};

export const logOut = () => {
  authentication.signOut();
}

function decodeToken() {
  const token = authentication.getAccessToken();
  const decoded: DecodedToken = jwtDecode(token);
  return decoded;
}

