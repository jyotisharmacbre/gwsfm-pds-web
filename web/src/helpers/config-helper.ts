const appConfig = () => {
    const config: any = (window as any).Config;
    if(config && config.REACT_APP_AUTH_INSTANCE==="#{AUTH_INSTANCE}#"){
      return {
        ...process.env
      }
    }
    return {
      ...config
    }
  };
  
  export default appConfig;