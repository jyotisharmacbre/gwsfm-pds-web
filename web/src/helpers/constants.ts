export const selectionButtons = [
  {
    title: 'BUTTON_YES',
    value: true
  },
  {
    title: 'BUTTON_NO',
    value: false
  }
];

export enum Notify {
  none,
  success,
  error
}
