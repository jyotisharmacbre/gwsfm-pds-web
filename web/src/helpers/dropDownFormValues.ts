export const projectStatusData: object[] = [
  { label: '1', value: '1' },
  { label: '2', value: '2' },
  { label: '3', value: '3' }
];

export const engagementData: object[] = [
  { label: 'Type 1', value: '1' },
  { label: 'Type 2', value: '2' },
  { label: 'Type 3', value: '3' }
];

export const typesOfEngagement: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'Fixed Price',
    value: 'Fixed Price'
  },
  {
    label: 'Time and Materials',
    value: 'Time and Materials'
  },
  {
    label: 'Risk and Reward',
    value: 'Risk and Reward'
  },
  {
    label: 'Cost Plus',
    value: 'Cost Plus'
  }
];

export const contractType: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'CT 1',
    value: 'CT 1'
  },
  {
    label: 'CT 2',
    value: 'CT 2'
  }
];

export const currency: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'AED (UAE Dirham)',
    value: 'AED'
  },
  {
    label: 'AFN (Afghani)',
    value: 'AFN'
  },
  {
    label: 'ALL (Lek)',
    value: 'ALL'
  },
  {
    label: 'AMD (Dram)',
    value: 'AMD'
  },
  {
    label: 'ANG (Dutch Antilles Guilder)',
    value: 'ANG'
  },
  {
    label: 'AOA (Kwanza)',
    value: 'AOA'
  },
  {
    label: 'ARS (Peso)',
    value: 'ARS'
  }
];

export const projectStatus: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'Initial Customer Inquiry',
    value: 'Initial Customer Inquiry'
  },
  {
    label: 'J&A',
    value: 'J&A'
  },
  {
    label: 'Archived',
    value: 'Archived'
  },
  {
    label: 'Lost',
    value: 'Lost'
  },
  {
    label: 'Order Received',
    value: 'Order Received'
  },
  {
    label: 'CPP',
    value: 'CPP'
  },
  {
    label: 'Complete',
    value: 'Complete'
  }
];

export const assetsWorkedOn: object[] = [
  {
    label: '---Please choose an option---',
    value: ''
  },
  {
    label: 'AS 1',
    value: 'AS 1'
  },
  {
    label: 'AS 2',
    value: 'AS 2'
  }
];

export const enquiryTypeData: object[] = [
  { label: 'Written', value: '1' },
  { label: 'Verbal', value: '2' },
  { label: 'Full tender enquiry with specifications & orgs', value: '3' },
  { label: 'Other', value: '4' }
];
