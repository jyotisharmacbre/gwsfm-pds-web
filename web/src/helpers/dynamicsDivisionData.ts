export const dynamicsDivisions = [
    {
      "DivisionId": "Name",
      "Description": "Name Level",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "CORPORATE",
      "Description": "Corporate Services",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "CRITICAL",
      "Description": "Critical EMEA",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "EXCEPTION",
      "Description": "Exceptional",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "HEADOFF",
      "Description": "Head Office",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "MANAGED",
      "Description": "Managed Services",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "EMEASERV",
      "Description": "EMEA Services",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "Name",
      "Description": "Name Level",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "CORPORATE",
      "Description": "Corporate Services",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "CRITICAL",
      "Description": "Critical Engineering",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "EXCEPTION",
      "Description": "Exceptional",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "HEADOFF",
      "Description": "Head Office",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "MANAGED",
      "Description": "Managed Services",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "EMEASERV",
      "Description": "EMEA Services",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "HEADOFF",
      "Description": "Head Office",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "MANAGED",
      "Description": "Managed Services",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "USHEADOFFICE",
      "Description": "US Head Office",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "UNITEDSTATES",
      "Description": "United States",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "SINGAPORE",
      "Description": "Singapore",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "FACILITIES",
      "Description": "Facilities Management Services",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "MAJOR",
      "Description": "Major Accounts Services & Europe",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "NATIONAL",
      "Description": "National Accounts",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "SWITZERLAND",
      "Description": "Switzerland",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "CRITICAL",
      "Description": "Critical Environment",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "GERMANY",
      "Description": "Germany",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "NATIONAL",
      "Description": "National Accounts",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "CRITICAL",
      "Description": "Critical Environment",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "NATIONAL",
      "Description": "National Accounts",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "POLAND",
      "Description": "Poland",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "CRITICAL",
      "Description": "Critical Environment",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "HONG KONG",
      "Description": "Hong Kong",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "National",
      "Description": "National Accounts",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "HONGKONG",
      "Description": "Hong Kong",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "NETHERLANDS",
      "Description": "Netherlands",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "REGCRIT",
      "Description": "Critical Regional",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "UKCRITICAL",
      "Description": "Critical UK",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "FACILTIES",
      "Description": "Facilities Management",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "National",
      "Description": "National Accounts",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "FRANCE",
      "Description": "France",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "NATIONAL",
      "Description": "NATIONAL",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "BELGIUM",
      "Description": "Belgium",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "NATIONAL",
      "Description": "NATIONAL",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "DATACENTRE",
      "Description": "Data Centre Solutions",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "FINANCIAL",
      "Description": "Financial Markets",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "FACILITIES",
      "Description": "FACILITIES",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "NATIONAL",
      "Description": "National",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "SWEDEN",
      "Description": "Sweden",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "FACILITIES",
      "Description": "Facilities Management",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "FACILITIES",
      "Description": "Facilities",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "FACILITIES",
      "Description": "Facilities Management",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "DATACENTRE",
      "Description": "Data Centre Solutions",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "FACILITIES",
      "Description": "Facilities",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "ENGSERV",
      "Description": "Engeering Services",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "SCOTIRL",
      "Description": "Scotland and Ireland",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "DATACENTRE",
      "Description": "Data Centre Solutions",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "DATACENTRE",
      "Description": "Data Centre Solutions",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "NATIONAL",
      "Description": "National",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "SCO-IRE",
      "Description": "Scotland & Ireland",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "NATIONAL",
      "Description": "National",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "SCO-IRL",
      "Description": "Scotland and Ireland",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "ITALY",
      "Description": "ITALY",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "SCO-IRL",
      "Description": "Scotland and Ireland",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "SCO-IRL",
      "Description": "Scotland and Ireland",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "NATIONAL",
      "Description": "National",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "SCO-IRL",
      "Description": "Scotland and Ireland",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "MULTI-SITE",
      "Description": "Emea Services",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "DATACENTRE",
      "Description": "Data Centre Solutions",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "DATACENTRE",
      "Description": "Data Centre Solutions",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "HEALTHCARE",
      "Description": "Healthcare",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "MULTI-SITE",
      "Description": "EMEA Services",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "SCOTIRL",
      "Description": "Scotland and Ireland",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "DATACENTRE",
      "Description": "Datacentre",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "BeneFraLux",
      "Description": "BeneFraLux",
      "DataSource": "External  EMEA_Acquired_Companies"
    },
    {
      "DivisionId": "BeneFraLux",
      "Description": "BeneFraLux",
      "DataSource": "External  EMEA_Acquired_Companies"
    },
    {
      "DivisionId": "BeneFraLux",
      "Description": "BeneFraLux",
      "DataSource": "External  EMEA_Acquired_Companies"
    },
    {
      "DivisionId": "BeneFraLux",
      "Description": "BeneFraLux",
      "DataSource": "External  EMEA_Acquired_Companies"
    },
    {
      "DivisionId": "C95002-Belgium",
      "Description": "Belgium",
      "DataSource": "External  EMEA_Acquired_Companies"
    },
    {
      "DivisionId": "BT",
      "Description": "BT",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "BT",
      "Description": "BT",
      "DataSource": "Dynamics"
    },
    {
      "DivisionId": "Retail",
      "Description": "RETAIL - Retail",
      "DataSource": "Dynamics"
    }
  ]