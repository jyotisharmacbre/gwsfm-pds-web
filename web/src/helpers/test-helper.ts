import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
