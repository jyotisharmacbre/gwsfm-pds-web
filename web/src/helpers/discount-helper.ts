import DiscountType from '../enums/DiscountType';
import { restrictMinusAndAllowDecimalForMaxRangeHundred, restrictMinusAndAllowDecimal } from '../helpers/utility-helper';

const normalizeClientDiscount = (discountType: DiscountType, discountValue) => {
    let updatedDiscount;
    if (discountType === DiscountType.Percent) {
        updatedDiscount = restrictMinusAndAllowDecimalForMaxRangeHundred(discountValue);
    }
    else updatedDiscount = restrictMinusAndAllowDecimal(discountValue);
    return updatedDiscount;
}

export default normalizeClientDiscount;