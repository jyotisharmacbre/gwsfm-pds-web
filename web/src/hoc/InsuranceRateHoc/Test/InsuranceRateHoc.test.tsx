import React from 'react';
import { mount } from 'enzyme';
import { findByTestAtrr } from '../../../helpers/test-helper';
import { insuranceRateHoc } from '../index';

let wrapper;

const mountComponent = (Props) => {
	const Component = (props) => <h1 data-test="insurance-rate">{props.insuranceRate}</h1>;
	const hoc = () => insuranceRateHoc(Component);
	const ComponentWithHOC = hoc();
	wrapper = mount(<ComponentWithHOC {...Props} />);
};
describe('Insurance rate HOC test cases', () => {
	const Props = {
		adminDefaultValues: [ { projectParameterId: 428, countryId: 143, name: 'Insurance_Rate_Perc', value: '2.6' } ]
	};
	beforeEach(() => {
		mountComponent(Props);
	});

	it('defines the component', () => {
		expect(wrapper).toBeDefined();
	});
	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should return the value 2.6 in insurance rate', () => {
		expect(findByTestAtrr(wrapper, 'insurance-rate').text()).toEqual('2.6');
	});
	it('should return the value 1.6 in insurance rate', () => {
		Props.adminDefaultValues[0].value = 1.6;
		mountComponent(Props);
		expect(findByTestAtrr(wrapper, 'insurance-rate').text()).toEqual('1.6');
	});
});
