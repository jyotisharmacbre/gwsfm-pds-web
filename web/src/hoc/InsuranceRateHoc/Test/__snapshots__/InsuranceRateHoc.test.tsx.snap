// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Insurance rate HOC test cases should match the snapshot 1`] = `ReactWrapper {}`;
